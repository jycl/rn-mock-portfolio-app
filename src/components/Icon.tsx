import React, { ReactElement } from 'react';
import { Image, StyleSheet } from 'react-native';
import Assets from '../assets';

type PropsType = {
  name: string;
  color: string;
};

/**
 * Icon displayed in bottom tab bar.
 * @param name key used to map to image
 * @param color tint color of image (changes when tab is focused/selected)
 */
const Icon = ({ name, color }: PropsType): ReactElement => {
  return (
    <Image style={[styles.icon, { tintColor: color }]} source={Assets[name]} />
  );
};

export default Icon;

const styles = StyleSheet.create({
  icon: {
    width: 28,
    height: 28,
  },
});
