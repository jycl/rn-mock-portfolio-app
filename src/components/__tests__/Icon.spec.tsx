import React from 'react';
import TestRenderer from 'react-test-renderer';
import Icon from '../Icon';

describe('Icon component', () => {
  it('should match snapshot rendering given defined props', () => {
    const renderItem = TestRenderer.create(<Icon name="more" color="black" />);
    expect(renderItem.toJSON()).toMatchSnapshot();
  });
});
