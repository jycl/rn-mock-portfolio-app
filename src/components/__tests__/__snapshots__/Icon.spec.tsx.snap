// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should match snapshot rendering given defined props 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/more.png",
    }
  }
  style={
    Array [
      Object {
        "height": 28,
        "width": 28,
      },
      Object {
        "tintColor": "black",
      },
    ]
  }
/>
`;
